backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/serialization/SerializeDescriptorReference.kt

```diff
@@ -0,0 +1,106 @@
+
+        if (descriptor is ParameterDescriptor || (descriptor is VariableDescriptor && descriptor !is PropertyDescriptor) || descriptor is TypeParameterDescriptor) return null
+
+        val containingDeclaration = descriptor.containingDeclaration!!
+
+        val (packageFqName, classFqName) = when (containingDeclaration) {
+            is ClassDescriptor -> {
+                val classId = containingDeclaration.classId ?: return null
+                Pair(classId.packageFqName.toString(), classId.relativeClassName.toString())
+            }
+            is PackageFragmentDescriptor -> Pair(containingDeclaration.fqName.toString(), "")
+            else -> return null
+        }
+
+        val isAccessor = declaration is IrSimpleFunction && declaration.correspondingProperty != null
+        val isBackingField = declaration is IrField && declaration.correspondingProperty != null
+        val isFakeOverride = declaration.origin == IrDeclarationOrigin.FAKE_OVERRIDE
+        val isDefaultConstructor =
+            descriptor is ClassConstructorDescriptor && containingDeclaration is ClassDescriptor && containingDeclaration.kind == ClassKind.OBJECT
+        val isEnumEntry = descriptor is ClassDescriptor && descriptor.kind == ClassKind.ENUM_ENTRY
+        val isEnumSpecial = declaration.origin == IrDeclarationOrigin.ENUM_CLASS_SPECIAL_MEMBER
+
+
+        val realDeclaration = if (isFakeOverride) {
+            when (declaration) {
+                is IrSimpleFunction -> declaration.resolveFakeOverrideMaybeAbstract()
+                is IrField -> declaration.resolveFakeOverrideMaybeAbstract()
+                is IrProperty -> declaration.resolveFakeOverrideMaybeAbstract()
+                else -> error("Unexpected fake override declaration")
+            }
+        } else {
+            declaration
+        }
+
+        val discoverableDescriptorsDeclaration: IrDeclaration? = if (isAccessor) {
+            (realDeclaration as IrSimpleFunction).correspondingProperty!!
+        } else if (isBackingField) {
+            (realDeclaration as IrField).correspondingProperty!!
+        } else if (isDefaultConstructor || isEnumEntry) {
+            null
+        } else {
+            realDeclaration
+        }
+
+        val uniqId = discoverableDescriptorsDeclaration?.let { declarationTable.uniqIdByDeclaration(it) }
+        uniqId?.let { declarationTable.descriptors.put(discoverableDescriptorsDeclaration.descriptor, it) }
+
+        val proto = KonanIr.DescriptorReference.newBuilder()
+            .setPackageFqName(packageFqName)
+            .setClassFqName(classFqName)
+            .setName(descriptor.name.toString())
+
+        if (uniqId != null) proto.setUniqId(protoUniqId(uniqId))
+
+        if (isFakeOverride) {
+            proto.setIsFakeOverride(true)
+        }
+
+        if (isBackingField) {
+            proto.setIsBackingField(true)
+        }
+
+        if (isAccessor) {
+            if (declaration.name.asString().startsWith("<get-"))
```

Comment:
It seems to be more reliable to compare `declaration` with `declaration.correspondingProperty.getter`/`.setter`.

Reply:
done